### PR TITLE
feat(Pin numpy version) New numpy version breaks a test

### PR DESCRIPTION
**Pull request by peterdeme**
As part of Python 2 to 3 migration project.

- Created virtualenvs for python 2.7.4, 3.7.16, 3.8.1: all tests succeeded for all versions.
- Numpy 1.18.0 breaks one of the tests: https://github.com/Censio/filterpy/blob/master/filterpy/kalman/tests/test_ukf.py#L56

The third parameter is a float but it should be an int. If I make it an int, the test fails, so I didn't know what to do

### CODE REVIEW

**Comment by burakerem** [comment]
`np.linspace(0,5,0.1)`

this should return an empty array. 

image

**Comment by burakerem** [comment]
I think the test fails when you change it to an int because it actually runs. If it's empty the assert never runs. I think that's supposed to be np.arange instead of np.linspace

**Comment by peterdeme** [comment]
@burakerem you are right, and not only that, but UKF doesn't even have a property "weights"

image

Do you have any idea what the author wanted to do here? If not, I'll remove the test

**Comment by burakerem** [comment]
These types of Kalman filter variations often involve a weighted average of sample points, whether it's Julier or Unscented or whatever. This appears to be a test that the weights generated by this particular filter for the weighted average sum to one, i.e. that they are a valid probability mass function (PMF). This ensures that you don't have to divide by this sum in every step.

**Comment by burakerem** [comment]
I think it's actually trying to test something like this:
https://github.com/Censio/filterpy/blob/master/filterpy/kalman/sigma_points.py#L172
I don't know what's going on here, tbh. I'm sight-reading this and it says it's testing Julier weights and then it uses the non-existent weights method of the Unscented Kalman Filter class.

Actually git blame says this was there before. What a weird change.
```def test_julier_weights():
    for n in range(1,15):
        for k in np.linspace(0,5,0.1):
            jp = JulierPoints(n, k)

            assert abs(sum(jp.Wm) - 1) < 1.e-12
            assert abs(sum(jp.Wc) - 1) < 1.e-12
```

**Review by yahehe — APPROVED** [review]

**Comment by peterdeme** [comment]
Leaving the PR open until @egelberg or @sanujitsahoo figures out whether we can support the new numpy version

**Comment by peterdeme** [comment]
Eddie helped me to fix the test

File: filterpy/kalman/tests/test_ukf.py [hunk 1]
```diff
@@ -52,11 +52,13 @@ def test_sigma_plot():
     assert max(Xi1[:,1]) > max(Xi0[:,1])
 
 def test_julier_weights():
-    for n in range(1,15):
-        for k in np.linspace(0,5,0.1):
-            Wm = UKF.weights(n, k)
-
+    for n in range(1, 15):
```
Comment (egelberg) [inline filterpy/kalman/tests/test_ukf.py:55]: :+1:

**Review by egelberg — APPROVED** [review]
Solid

**Pull request merged**